From 22f9b4c825512c8c530c7f2c78607e6c016dcc61 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:26 -0400
Subject: [fs] btrfs: use right type to get real comparison

Message-id: <1407179109-13714-380-git-send-email-sandeen@redhat.com>
Patchwork-id: 87321
O-Subject: [RHEL7.1 PATCH 379/422] Btrfs: use right type to get real comparison
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Liu Bo <bo.li.liu@oracle.com>

commit cd857dd6bc2ae9ecea14e75a34e8a8fdc158e307
Author: Liu Bo <bo.li.liu@oracle.com>
Date:   Sun Jun 8 19:04:13 2014 +0800

    Btrfs: use right type to get real comparison

    We want to make sure the point is still within the extent item, not to verify
    the memory it's pointing to.

    Signed-off-by: Liu Bo <bo.li.liu@oracle.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/backref.c b/fs/btrfs/backref.c
index 84d0912..00f9d9f 100644
--- a/fs/btrfs/backref.c
+++ b/fs/btrfs/backref.c
@@ -1428,7 +1428,7 @@ static int __get_extent_inline_ref(unsigned long *ptr, struct extent_buffer *eb,
    *out_eiref = (struct btrfs_extent_inline_ref *)(ei + 1);
   }
   *ptr = (unsigned long)*out_eiref;
-  if ((void *)*ptr >= (void *)ei + item_size)
+  if ((unsigned long)(*ptr) >= (unsigned long)ei + item_size)
    return -ENOENT;
  }
 
-- 
1.7.1